commit 1bda8b7996656c85005db7ac2a86b797c7f6cd1d
Author: Christoph Hellwig <hch@lst.de>
Date:   Tue Nov 20 16:57:54 2018 +0100

    nvme: warn when finding multi-port subsystems without multipathing enabled
    
    [ Upstream commit 14a1336e6fff47dd1028b484d6c802105c58e2ee ]
    
    Without CONFIG_NVME_MULTIPATH enabled a multi-port subsystem might
    show up as invididual devices and cause problems, warn about it.
    
    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Reviewed-by: Sagi Grimberg <sagi@grimberg.me>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/nvme/host/nvme.h b/drivers/nvme/host/nvme.h
index bb4a2003c097..60220de2db52 100644
--- a/drivers/nvme/host/nvme.h
+++ b/drivers/nvme/host/nvme.h
@@ -537,6 +537,9 @@ static inline void nvme_mpath_check_last_path(struct nvme_ns *ns)
 static inline int nvme_mpath_init(struct nvme_ctrl *ctrl,
 		struct nvme_id_ctrl *id)
 {
+	if (ctrl->subsys->cmic & (1 << 3))
+		dev_warn(ctrl->device,
+"Please enable CONFIG_NVME_MULTIPATH for full support of multi-port devices.\n");
 	return 0;
 }
 static inline void nvme_mpath_uninit(struct nvme_ctrl *ctrl)
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1119_1bda8b7996656c85005db7ac2a86b797c7f6cd1d.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1119_1bda8b7996656c85005db7ac2a86b797c7f6cd1d.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
